src/main/java/java/lang/String.java

```diff
@@ -1390,11 +1390,12 @@ else if(c<=0xFFFF)
      * @see  #equalsIgnoreCase(String)
      *
      * @diffblue.fullSupport
-     * @diffblue.untested
      */
     public boolean equals(Object anObject) {
-        // DIFFBLUE MODEL LIBRARY This is treated internally in CBMC
-        return CProver.nondetBoolean();
+        if (anObject instanceof String) {
```

Comment:
As in the models-lib PR, could you add the original code from the JDK, commented out?